src/parse_instruction.py

```diff
@@ -232,11 +232,32 @@ class NaiveParsingArch(ArchAsmParsing):
     machinery to reduce arch dependence."""
 
     all_regs: List[Register] = []
-    aliased_regs: Dict[Register, Register] = {}
+    aliased_regs: Dict[str, Register] = {}
 
     def normalize_instruction(self, instr: AsmInstruction) -> AsmInstruction:
         return instr
 
+@dataclass
+class RegFormatter:
+    """Converts register names used in input assembly to the internal register representation, 
+    saving the original names for use in the output"""
```

Comment:
A bit focussed on the input parsing part, which is a bit weird given the name of the class. Shrug